drivers/can/can_mcp251xfd.c

```diff
@@ -978,12 +979,86 @@ static int mcp251xfd_handle_ivmif(const struct device *dev)
 	reg = mcp251xfd_get_spi_buf_ptr(dev);
 	reg[0] = 0;
 	ret = mcp251xfd_write(dev, MCP251XFD_REG_BDIAG1, MCP251XFD_REG_SIZE);
+	if (ret < 0) {
+		goto done;
+	}
+
+	/* There's no flag for DACKERR */
+	if ((tmp & MCP251XFD_REG_BDIAG1_NACKERR) != 0) {
+		CAN_STATS_ACK_ERROR_INC(dev);
+		LOG_INF("setting ack error nom");
+	}
+
+	if ((tmp & MCP251XFD_REG_BDIAG1_NBIT0ERR) != 0) {
+		CAN_STATS_BIT0_ERROR_INC(dev);
+	}
+
+	if ((tmp & MCP251XFD_REG_BDIAG1_NBIT1ERR) != 0) {
+		CAN_STATS_BIT1_ERROR_INC(dev);
+	}
+
+	if ((tmp & MCP251XFD_REG_BDIAG1_DBIT0ERR) != 0) {
+		CAN_STATS_BIT0_ERROR_INC(dev);
+	}
+
+	if ((tmp & MCP251XFD_REG_BDIAG1_DBIT1ERR) != 0) {
+		CAN_STATS_BIT1_ERROR_INC(dev);
+	}
+
+	if ((tmp & MCP251XFD_REG_BDIAG1_NCRCERR) != 0) {
+		CAN_STATS_CRC_ERROR_INC(dev);
+	}
+
+	if ((tmp & MCP251XFD_REG_BDIAG1_DCRCERR) != 0) {
+		CAN_STATS_CRC_ERROR_INC(dev);
+	}
```

Comment:
Can the bits causing the same error counters to be increment be combined? Example:

```suggestion
	if ((tmp & (MCP251XFD_REG_BDIAG1_NCRCERR | MCP251XFD_REG_BDIAG1_DCRCERR)) != 0) {
		CAN_STATS_CRC_ERROR_INC(dev);
	}
```

Reply:
Yes, I just wasn't sure if both errors could be set at once which cause errors in the counts.